src/Nethermind/Nethermind.Optimism/Rpc/OptimismTransactionForRpc.cs

```diff
@@ -3,27 +3,96 @@
 
 using Nethermind.Core.Crypto;
 using Nethermind.Core;
-using Nethermind.Facade.Eth;
 using Nethermind.Int256;
 using System.Text.Json.Serialization;
+using Nethermind.Facade.Eth.RpcTransaction;
+using Nethermind.Facade.Eth;
 
 namespace Nethermind.Optimism.Rpc;
 
+/// <Remarks>
+/// Defined in https://github.com/ethereum-optimism/op-geth/blob/8af19cf20261c0b62f98cc27da3a268f542822ee/core/types/deposit_tx.go#L29-L46
+/// </Remarks>
 public class OptimismTransactionForRpc : TransactionForRpc
 {
-    public OptimismTransactionForRpc(Hash256? blockHash, OptimismTxReceipt? receipt, Transaction transaction, UInt256? baseFee = null)
-       : base(blockHash, receipt?.BlockNumber, receipt?.Index, transaction, baseFee)
-    {
-        if (transaction.Type == TxType.DepositTx)
-        {
-            SourceHash = transaction.SourceHash;
-            Mint = transaction.Mint;
-            IsSystemTx = transaction.IsOPSystemTransaction ? true : null;
-            Nonce = receipt?.DepositNonce;
-            DepositReceiptVersion = receipt?.DepositReceiptVersion;
-        }
-    }
+    public override TxType? Type => TxType.DepositTx;
+
+    public Hash256 SourceHash { get; set; }
+
+    public Address From { get; set; }
+
+    [JsonIgnore(Condition = JsonIgnoreCondition.Never)]
+    public Address? To { get; set; }
+
+    public UInt256? Mint { get; set; }
 
+    public UInt256 Value { get; set; }
+
+    public ulong Gas { get; set; }
+
+    public bool IsSystemTx { get; set; }
+
+    public byte[] Input { get; set; }
+
+    public UInt256? Nonce { get; set; }
+
+    #region Nethermind specific fields
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
     public UInt256? DepositReceiptVersion { get; set; }
+    #endregion
+
+#pragma warning disable CS8618
+    [JsonConstructor]
+    public OptimismTransactionForRpc() { }
+#pragma warning restore CS8618
```

Comment:
Several fields are required based on the original Optimism implementation (see: https://github.com/ethereum-optimism/op-geth/blob/60038121c7571a59875ff9ed7679c48c9f73405d/core/types/transaction_marshalling_test.go#L27-L68) but we we're accepting null/missing values.

Should we continue accepting these transactions or should we throw when required fields are missing?

Reply:
Changed to throwing when required fields are missing.